frontend/src/views/OrderView.vue

```diff
@@ -1,74 +1,183 @@
-<script setup>
+<script setup lang="ts">
+import AutoComplete from "primevue/autocomplete";
 import Button from "primevue/button";
+import Card from "primevue/card";
 import Dropdown from "primevue/dropdown";
 import FloatLabel from "primevue/floatlabel";
+import InputNumber from "primevue/inputnumber";
 import InputText from "primevue/inputtext";
 import Textarea from "primevue/textarea";
-import { ref } from "vue";
+import { computed, onMounted, ref } from "vue";
+import { useRoute, useRouter } from "vue-router";
 
+import { createOrder, getClients } from "@/backendClient";
 import { OrderStatus } from "@/global/types/appTypes";
+import type { OrderCreate, OrderUpdate } from "@/global/types/dataEditTypes";
+import type { Client } from "@/global/types/entities";
+import { ROUTES } from "@/router";
+
+interface ExtendedClient extends Client {
+  full_name?: string;
+}
+let orderInfo = ref<OrderUpdate | OrderCreate>({
+  client_id: "",
+  status: OrderStatus.preparation,
+  title: "",
+  description: "",
+  discount_percentage: 0,
+  discount_duration: 7,
+  can_have_cash_discount: false,
+});
 
 const orderStatusTypes = Object.values(OrderStatus);
-let status = ref(OrderStatus.preparation);
 
-const discountChoice = ["ja", "nein"];
-let discount = ref(discountChoice[0]);
+const discountChoice = [
+  { value: true, label: "ja" },
+  { value: false, label: "nein" },
+];
+const isEditing = computed(() => {
+  return Boolean(route.params.id);
+});
+const discountPeriodChoice = [7, 14];
+
+const selectedClient = ref<ExtendedClient>();
+
+const filteredClients = ref<ExtendedClient[]>([]);
+const clientsList = ref<ExtendedClient[]>([]);
+
+const isButtonDisabled = computed(() => {
+  if (orderInfo.value.title && selectedClient.value?.id && orderInfo.value.description) {
+    return false;
+  } else {
+    return true;
+  }
+});
+
+const searchClient = (event: any) => {
+  setTimeout(() => {
+    if (!event.query.trim().length) {
+      filteredClients.value = [...clientsList.value];
+    } else {
+      filteredClients.value = clientsList.value.filter((client: Client) => {
+        return (
+          client.last_name?.toLowerCase().startsWith(event.query.toLowerCase()) ||
+          client.first_name?.toLowerCase().startsWith(event.query.toLowerCase())
+        );
+      });
+    }
+  }, 250);
+};
+const router = useRouter();
+const route = useRoute();
+function onOrdersList() {
+  router.push(`${ROUTES.ORDER.path}`);
+}
 
-const discountPeriodChoice = ["7", "14"];
-let discountPeriod = ref(discountPeriodChoice[0]);
+const onSaveOrder = async () => {
```

Comment:
If it is a function, why not use a normal definition?

```suggestion
async function onSaveOrder () {
```

Reply:
It's just another way of declaring function. Is it important to use only one way? I can modify it in the next pull request if it necessary.

Reply:
> Is it important to use only one way? 

Consistency is good to avoid confusion later on. We can certainly fix it in the following pull request, though 👌🏻